### PR TITLE
Linked hash map

### PR DESCRIPTION
**Pull request by ruslansennov**

### CODE REVIEW

**Comment by ruslansennov** [comment]
#612

File: src/main/java/javaslang/collection/HashSet.java [hunk 1]
```diff
@@ -800,7 +800,7 @@ public boolean equals(Object o) {
 
     @Override
     public String toString() {
-        return mkString(", ", "HashSet(", ")");
+        return mkString("HashSet(", ", ", ")");
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/HashSet.java]: Oh, thanks for catching - I thought I've catched all after changing `mkString(infix, prefix, suffix)` to `mkString(prefix, infix, suffix)`.

**Comment by danieldietrich** [comment]
Wow, THAT. WAS. FAST. Thx!

**Comment by danieldietrich** [comment]
Fixes #612

File: src/main/java/javaslang/collection/HashSet.java (hunk 1 above)
Comment (ruslansennov) [inline src/main/java/javaslang/collection/HashSet.java]: that PR was very dangerous :smile:

**Pull request merged**